group-generators/helpers/data-providers/gitpoap/index.ts

```diff
@@ -0,0 +1,49 @@
+import axios from "axios";
+import {  GitPoapAddresses } from "./types";
+import { FetchedData } from "topics/group";
+
+export class GitPoapProvider {
+  url: string;
+
+
+  public constructor() {
+    this.url = "https://public-api.gitpoap.io/";
+  }
+
+  private async getGitPoap(endpoint: string): Promise<any> {
+    const { data: res } = await axios({
+      url: this.url + endpoint, 
+      method: "get",
+    });
+    return res;
+  }
+
+
+  public async getGitPoapHoldersByEventId(gitPoapEventId: string ): Promise<FetchedData> {
```

Comment:
This will not work in the Factory

Because when the group is generated through the Factory, it will put all the argument in an object like this:
`getGitPoapHoldersByEventId({gitPoapEventId,...})`

So you need to create an object gitPoapEventId like this:
`getGitPoapHoldersArg = {gitPoapEventId: string}`

You can inspire you of this dataProvider for example: https://github.com/sismo-core/sismo-hub/tree/main/group-generators/helpers/data-providers/talentlayer

Reply:
thanks fixed with https://github.com/sismo-core/sismo-hub/pull/1555/commits/8b21fef174ab9ed3a5b3b92a8535794cba850c46